dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceCache.java

```diff
@@ -0,0 +1,220 @@
+     * RuntimeServices)
+     */
+    public void initialize(RuntimeServices rs) {
+        cache = CacheLocator.getCacheAdministrator();
+    }
+
+    public void addMiss(Object resourceKey) {
+        Logger.info(this.getClass(), "velocityMiss:" + resourceKey);
+    }
+
+    public boolean isMiss(Object resourceKey) {
+        return false;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#put(java.lang.Object,
+     * org.apache.velocity.runtime.resource.Resource)
+     */
+    public Resource put(Object resourceKey, Resource resource) {
+        if (resource != null && ignoreGlobalVM.contains(resource.getName())) {
+            return resource;
+        }
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+
+
+        String key = group + cleanedResourceKey;
+        // Add the key to the cache
+        cache.put(key, resource, group);
+
+        return resource;
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#remove(java.lang.Object)
+     */
+    public Resource remove(Object resourceKey) {
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+
+        String key = group + cleanedResourceKey;
+
+        try {
+            cache.remove(key, group);
+        } catch (Exception e) {
+            Logger.debug(this, e.getMessage(), e);
+        }
+        return null;
+    }
+
+    public void clearCache() {
+        for (String group : groupNames) {
+            cache.flushGroup(group);
+        }
+
+    }
+
+    @Deprecated
+    public void clearMenuCache() {
+
+    }
+
+    public String[] getGroups() {
+        return groupNames;
+    }
+
+    public String getPrimaryGroup() {
+        return primaryGroup;
+    }
+
+    private String cleanKey(String key) {
+        if (key.startsWith(ResourceManager.RESOURCE_TEMPLATE + ""))
+            key = key.substring((ResourceManager.RESOURCE_TEMPLATE + "").length());
+
+        if (key.startsWith(File.separatorChar + "")) {
+            key = key.substring(1);
+        }
+        if (key.startsWith("/")) {
+            key = key.substring(1);
+        }
+        key = StringUtil.replace(key, '\\', '/');
+        return key;
+    }
+
+    @Deprecated
+    public String getMenuGroup() {
+        return primaryGroup;
+    }
+
+    public void removeContentTypeFile(ContentType contentType) {
+        String folderPath = "working/";
+        String filePath = folderPath + contentType.inode() + "." + VelocityType.CONTENT_TYPE.fileExtension;
+
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        String absolutPath = velocityRootPath + File.separator + filePath;
+        java.io.File f = new java.io.File(absolutPath);
+        f.delete();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS899)